services/submission/src/main/java/app/coronawarn/server/services/submission/checkins/EventCheckinFacade.java

```diff
@@ -0,0 +1,88 @@
+import app.coronawarn.server.common.persistence.service.utils.checkins.CheckinsDateSpecification;
+import app.coronawarn.server.common.protocols.internal.SubmissionPayload;
+import app.coronawarn.server.common.protocols.internal.SubmissionPayload.SubmissionType;
+import app.coronawarn.server.common.protocols.internal.pt.CheckIn;
+import app.coronawarn.server.services.submission.config.SubmissionServiceConfig;
+import app.coronawarn.server.services.submission.controller.CheckinsStorageResult;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EventCheckinFacade {
+
+  private static final Logger logger = LoggerFactory.getLogger(EventCheckinFacade.class);
+
+  public static final Marker EVENT = MarkerFactory.getMarker("EVENT");
+
+  private TraceTimeIntervalWarningService traceTimeIntervalWarningService;
+  private FakeCheckinsGenerator fakeCheckinsGenerator;
+  private final EventCheckinDataFilter checkinsDataFilter;
+  private final SubmissionServiceConfig submissionServiceConfig;
+
+
+  /**
+   * Create a new instance.
+   */
+  public EventCheckinFacade(TraceTimeIntervalWarningService traceTimeIntervalWarningService,
+      FakeCheckinsGenerator fakeCheckinsGenerator,
+      EventCheckinDataFilter checkinsDataFilter,
+      SubmissionServiceConfig submissionServiceConfig) {
+    this.traceTimeIntervalWarningService = traceTimeIntervalWarningService;
+    this.fakeCheckinsGenerator = fakeCheckinsGenerator;
+    this.checkinsDataFilter = checkinsDataFilter;
+    this.submissionServiceConfig = submissionServiceConfig;
+  }
+
+  /**
+   * For each checkin in the given list, generate other fake checkin data based on the passed in number and store
+   * everything as {@link TraceTimeIntervalWarning} entities. Returns the number of inserted entities which is useful
+   * for the case where there might be conflicts with the table constraints during the db save operations.
+   */
+  public int saveCheckinsWithFakeData(List<CheckIn> originalCheckins, int numberOfFakesToCreate,
+      byte[] pepper, int submissionTimestamp, SubmissionType submissionType) {
+    List<CheckIn> allCheckins = new ArrayList<>(originalCheckins);
+    allCheckins.addAll(fakeCheckinsGenerator.generateFakeCheckins(originalCheckins,
+        numberOfFakesToCreate, pepper));
+    return traceTimeIntervalWarningService.saveCheckins(allCheckins, submissionTimestamp, submissionType);
+  }
+
+  /**
+   * Extract check-ins from submission payload and store them.
+   * @param submissionPayload - submission payload
+   * @return - storage result containing number of filtered and saved check-ins.
+   */
+  public CheckinsStorageResult extractAndStoreEventCheckins(SubmissionPayload submissionPayload) {
```

Comment:
Please make sure to test this method too :)

Reply:
Added.